### DISCUSSION TITLE
Do we have to train the pytorchclassifier estimator with training data every time before applying attacks?

### DISCUSSION THREAD

**Discussion by anukriti15** [Trusted-AI/adversarial-robustness-toolbox#2255] [Q&A]
Will the attacks not work if the user gives a saved model with weights, do we have to call classifier.fit() every time for it to work with attacks?

**Comment by beat-buesser** [comment]
Hi @anukriti15 No, classifier.fit() is not required, you can provide a pre-trained model to any `art.estimator`.

**Reply by anukriti15** [reply]
@beat-buesser I am not getting sufficient accuracy when I am doing classifier.predict() without the 'fit' function. The model's accuracy on my test samples is 73.45476% but when using ART Pytorch classifier estimator and passing the pre-trained model without fitting again, it gives an accuracy of 48.1910%. Do I have to do anything additional apart from torch.load(<saved model path>)?

**Reply by anukriti15** [reply]
# Loading the model 
model = torch.jit.load('SavedModel.pt')
print(model)
model.eval()
#optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
loss = nn.CrossEntropyLoss()

classifier = PyTorchClassifier(
    model=model,
    loss=loss,
    #optimizer=optimizer,
    input_shape=(3, 224, 224),
    nb_classes=5,
    #device_type='gpu'
)

# Fitting the model
#classifier.fit(x_train, y_train, batch_size=1, nb_epochs=10)

# Evaluate the ART classifier on benign test examples
predictions = classifier.predict(x_test, training_mode=False)
accuracy = np.sum(np.argmax(predictions, axis=1) == y_test) / len(y_test)
print("Accuracy on benign test examples: {}%".format(accuracy * 100))

This is a snippet of the code I am working on..
